### PR TITLE
Removed the closed handler that was just reconnecting Fixes #1150

### LINKED ISSUE #1150: TestClient Verify
It leaves its connections open after the command completes. Not sure if retries or? Note log (completed almost 22 hours earlier).

example:

|           46|   Success|
--------------------------
|           47|   Success|
--------------------------
|           48|   Success|
--------------------------
|           49|   Success|
--------------------------

[29970,05,23:21:28.694] Command exited with code 0.
>>> [29970,109,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {9612202e-736d-42ee-afa2-fed70c7b4d30}] failed. Error: ConnectionRefused.
[29970,84,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {1a927bbb-7463-4f3d-9fce-4c3050ebe04e}] failed. Error: ConnectionRefused.
[29970,97,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {0f0ea36e-690b-440d-82fc-d2deb97de3e8}] failed. Error: ConnectionRefused.
[29970,101,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {c00fb881-6fde-43d4-9eda-fa5516e34a19}] failed. Error: ConnectionRefused.
[29970,94,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {5f6bd3f1-636b-4a2b-94cd-222439ec2c3b}] failed. Error: ConnectionRefused.
[29970,113,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {996fac67-b778-4599-9bd9-9b603e73346d}] failed. Error: ConnectionRefused.
[29970,129,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {53c4c490-8f9b-44bc-9e2f-1bcf96209b6e}] failed. Error: ConnectionRefused.
[29970,105,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {fd6fcf3c-21c0-4115-afe2-6b280934dd06}] failed. Error: ConnectionRefused.
[29970,118,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {f91d5a05-b5f5-4fbb-8e72-fb85f2fcf0ce}] failed. Error: ConnectionRefused.
[29970,137,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {c44e4c6c-e432-4c5c-9e67-6c416b0ce48a}] failed. Error: ConnectionRefused.
[29970,131,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {f40b5361-5554-4fe8-b63e-0c93855d5d58}] failed. Error: ConnectionRefused.
[29970,115,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {aa55ead2-ce85-4a90-95e1-8d8a2ed6c1db}] failed. Error: ConnectionRefused.
[29970,107,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {001846fc-5f58-4faf-b9d4-ef85ec622498}] failed. Error: ConnectionRefused.
[29970,143,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {815741a7-5a4b-4b82-9f3f-ddb292dcdfbf}] failed. Error: ConnectionRefused.
[29970,144,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {81a5298d-e272-4f3d-9e82-71a71edb8041}] failed. Error: ConnectionRefused.
[29970,96,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {d3c6f35f-3698-4d3e-85ff-9fa7fe07bd8f}] failed. Error: ConnectionRefused.
[29970,138,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {6e3668fa-289d-4ae5-8a8c-6b0ecd66bee0}] failed. Error: ConnectionRefused.
[29970,85,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {4553c9e0-a86a-40fe-a1cf-d670fe1c3078}] failed. Error: ConnectionRefused.
[29970,116,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {eda3e40d-c737-42d7-9df9-c71a129f8efb}] failed. Error: ConnectionRefused.
[29970,99,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {bdf56cde-eec8-4260-917c-52b9a6bced40}] failed. Error: ConnectionRefused.
[29970,08,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {4d92c492-e160-44cb-8f33-e931a9b9a004}] failed. Error: ConnectionRefused.
[29970,89,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {9bb2a733-6b1e-4d81-92c0-ab08f8da4789}] failed. Error: ConnectionRefused.
[29970,100,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {56521bc6-3a8a-4ed6-8763-d6fd537e2875}] failed. Error: ConnectionRefused.
[29970,142,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {8f649d90-ff54-4e87-a1f6-8fbf5e32abde}] failed. Error: ConnectionRefused.
[29970,90,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {e03cfb53-93a0-4503-aa9a-51d7386d0ce3}] failed. Error: ConnectionRefused.
[29970,126,21:20:30.326] TcpTypedConnection: connection to [127.0.0.1:1113, L, {940e145d-b382-4326-b5f9-85b7e64163bf}] failed. Error: ConnectionRefused.

### PR DESCRIPTION
**Pull request by riccardone**
Fixes #1150

### CODE REVIEW

**Comment by riccardone** [comment]
@hayley-jean do you mean something like this:

```            
closed = (_, __) =>
            {
                if (!context.Client.Options.Reconnect) return;
                Thread.Sleep(TimeSpan.FromSeconds(1));
                connection = context.Client.CreateTcpConnection(context, packageHandler, cn => iteration.Set(), closed, false);
            };
```
and then:

```
context.Client.Options.Reconnect = false;
connection.Close();
```

**Comment by hayley-jean** [comment]
@riccardone Yeah, that looks good 👍

**Comment by riccardone** [comment]
@hayley-jean I applied the required changes

File: src/EventStore.TestClient/ClientOptions.cs [hunk 1]
```diff
@@ -56,6 +57,7 @@ public ClientOptions()
             WriteWindow = 2000;
             PingWindow = 2000;
             Force = false;
+            Reconnect = false;
```
Comment (hayley-jean) [inline src/EventStore.TestClient/ClientOptions.cs]: We need Reconnect to be true by default, and then to change it to false when we want to properly close the connection without reconnecting. We don't want to lose the reconnect behaviour while the test is running

So for example in the DvuBasicProcessor just before finishing Write():

```status.FinilizeStatus(writerIdx, failed != sent);
context.Client.Options.Reconnect = false;
connection.Close();
finish.Set();```

**Comment by hayley-jean** [comment]
Thanks @riccardone just one change that needs to be made, see the inline comment

I'm not sure what's up with wercker at the moment, but it's not your change failing, wercker's just having trouble checking out the PR.

**Comment by hayley-jean** [comment]
Looks good @riccardone 👍 

Please can you squash your commits, and we will merge after RC2 has been released

**Pull request closed without merging**